components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -183,10 +183,15 @@ pub struct DedicatedWorkerGlobalScope {
     #[ignore_malloc_size_of = "Arc"]
     image_cache: Arc<dyn ImageCache>,
     browsing_context: Option<BrowsingContextId>,
+    /// A receiver of control messages,
+    /// currently only used to signal shutdown.
+    #[ignore_malloc_size_of = "Channels are hard"]
+    control_receiver: Receiver<()>,
```

Comment:
Use a `ShutdownMsg` type rather than `()`?

Reply:
The idea is that the shutdown is signalled by dropping the sender(and to avoid explicit shutdown messages), and the `()` is used as a placeholder since not other messages are used for now. Re the sthutdown, I'll answer below.

Reply:
I also like using Never as a message type in these cases, which make it clear(I.e statically impossible) to send a message: https://github.com/rust-lang/rls/blob/17a439440e6b00b1f014a49c6cf47752ecae5bb7/rls/src/concurrency.rs#L88

It also removes the question of handling OK case on the receive side (you write an empty match: `match never {}`)

Admittedly, “wtf is Sender<Never>“ is a common reaction when people see such code for the first time. On the flip side, once they learn the pattern, `shutdown: Sender<Never>` becomes an immediately recognized code shape.

Reply:
I think that's a good idea, added a commit that does this...